src/master/master_impl.cc

```diff
@@ -2090,36 +2078,14 @@ void MasterImpl::TabletNodeRecoveryCallback(std::string addr,
     }
     node->ResetQueryFailCount();
 
-    // recovery offline tablets
-    // unload should-not-exist tablets
+    // New tabletnode should not have any tablet.
     uint32_t meta_num = response->tabletmeta_list().meta_size();
-    for (uint32_t i = 0; i < meta_num; i++) {
```

Comment:
Good point. Do you mind if I remove the query logic and call this function within AddTabletNode?